### PR TITLE
Feat/civ 1616 send notification noc

### PR DESCRIPTION
**Pull request by kurt-uk**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-1616

### Change description ###
Added callback handler for NOC former and other sol 1 and 2, will skip former sol if the case has LiP as there is no former sol to contact. 

Branched from 1610 new 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by HarryH96 — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/utils/NocNotificationUtils.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    private NocNotificationUtils() {
+        //NO-OP
+    }
+
+    public static String getPreviousSolicitorEmail(CaseData caseData) {
+        return caseData.getChangeOfRepresentation().getFormerRepresentationEmailAddress();
+    }
+
+    private static void setOtherSolicitor1(CaseData caseData) {
+        if (isRespondent2NewSolicitor(caseData)) {
```
Comment (sherlynkhaw) [inline src/main/java/uk/gov/hmcts/reform/civil/utils/NocNotificationUtils.java:22]: why is it that only when Respondent2 Solicitor is changed, we notify applicantsolicitor?

Shouldn't all be applicantsolicitor?

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/ChangeOfRepresentationNotificationHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+                .thenReturn(Optional.of(Organisation.builder().name(OTHER_SOLICITOR_2).build()));
+            when(notificationsProperties.getNoticeOfChangeFormerSolicitor()).thenReturn(PREVIOUS_SOL_TEMPLATE);
+            when(notificationsProperties.getNoticeOfChangeOtherParties()).thenReturn(OTHER_SOL_TEMPLATE);
+        }
+
+        @Nested
+        class NotifyPreviousSolicitorEvent {
+
+            @Test
+            void shouldNotifyFormerSolicitorWhenInvoked_whenInvoked() {
```
Comment (sherlynkhaw) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/ChangeOfRepresentationNotificationHandlerTest.java:92]: remove double when invoked?

**Review by sherlynkhaw — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/utils/NocNotificationUtils.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            otherSolicitor1Email = caseData.getApplicantSolicitor1UserDetails().getEmail();
+            otherSolicitor1OrgId = caseData.getApplicant1OrganisationPolicy().getOrganisation().getOrganisationID();
+        } else if (caseData.getChangeOfRepresentation().getCaseRole().equals(CaseRole.APPLICANTSOLICITORONE)) {
+            otherSolicitor1Email = caseData.getRespondentSolicitor2EmailAddress();
```
[the remark this answers is not in the document]
Comment (kurt-uk) [reply src/main/java/uk/gov/hmcts/reform/civil/utils/NocNotificationUtils.java]: we only get respondent 2 if the event is triggered for that 2nd solicitor

File: src/main/java/uk/gov/hmcts/reform/civil/utils/NocNotificationUtils.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            otherSolicitor1OrgId = caseData.getRespondent2OrganisationPolicy().getOrganisation().getOrganisationID();
+        } else {
```
[the remark this answers is not in the document]
Comment (kurt-uk) [reply src/main/java/uk/gov/hmcts/reform/civil/utils/NocNotificationUtils.java:29]: not sure what you mean, the logic for the scenarios were dicussed and that didnt come up

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/ChangeOfRepresentationNotificationHandlerTest.java (hunk 2 above)
Comment (kurt-uk) [reply src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/ChangeOfRepresentationNotificationHandlerTest.java]: done

File: src/main/java/uk/gov/hmcts/reform/civil/utils/NocNotificationUtils.java [hunk 5]
```diff
@@ -18,51 +18,95 @@ public static String getPreviousSolicitorEmail(CaseData caseData) {
...
+                return RecipientData.builder()
+                    .email(caseData.getRespondentSolicitor1EmailAddress())
+                    .orgId(caseData.getRespondent1OrganisationPolicy().getOrganisation().getOrganisationID())
+                    .build();
+            }
         } else {
-            otherSolicitor2Email = caseData.getRespondentSolicitor2EmailAddress();
-            otherSolicitor2OrgId = caseData.getRespondent2OrganisationPolicy().getOrganisation().getOrganisationID();
+            if (getMultiPartyScenario(caseData).equals(TWO_V_ONE)) {
+                return RecipientData.builder()
```
Comment (sherlynkhaw) [inline src/main/java/uk/gov/hmcts/reform/civil/utils/NocNotificationUtils.java:53]: hmmm, if it's a 2v1, do we need to notify applicantsolicitor1 again? i think it's worth a check with BAs/Leanne

**Review by sherlynkhaw — DISMISSED** [review]
I think it looks better now!
I would add some comments in your NocNotificationUtils.java, so it'll be easier to read/digest later on.

**Review by sankaviv1 — APPROVED** [review]

**Review by LTurkingtonHMCTS — DISMISSED** [review]

**Review by Gareth40343 — APPROVED** [review]

**Review by HarryH96 — APPROVED** [review]

**Review by LTurkingtonHMCTS — APPROVED** [review]

**Pull request merged**